### DISCUSSION TITLE
Custom validators for types from third party libraries

### DISCUSSION THREAD

**Discussion by anirban-ray-publicis-sapient** [pydantic/pydantic#5906] [Question] [answered]
For most of the libraries, they define their own classes. Let's assume the class name is `Custom`.

If they are added as type hints and `validate_arguments` is used, it fails with:

> RuntimeError: no validator found for <class 'Custom'>, see `arbitrary_types_allowed` in Config

What I ended up doing is that creating a custom type as mentioned in the docs:

```python
class CustomWithValidator(Custom):
    @classmethod
    def __get_validators__(cls: typing.Type["CustomWithValidator"]) -> collections.abc.Iterable[collections.abc.Callable]:
        yield cls.validate_custom

    @classmethod
    def validate_custom(cls: typing.Type["CustomWithValidator"], passed_value: typing.Any) -> Custom:
        if isinstance(passed_value, Custom):
            return passed_value

        raise ValueError
```

And then use `CustomWithValidator` as the type hint instead of `Custom`.

This works, but there are too many custom types and it's almost identical code for every single one of them. Is there a way to handle this in general for any type?

**Comment by adriangb** [comment, marked as answer]
In V2 this will be improved to:

```python
from lib import CustomType as CustomTypeLib
from pydantic.types import InstanceOf

CustomType = InstanceOf[CustomTypeLib]
```

Then `CustomType` can be used as a field on any model.

**Reply by anirban-ray-publicis-sapient** [reply]
Thank you. Interestingly, I searched the docs with `InstanceOf`, but found none even after toggling to `dev-v2`.

Is there a way to achieve similar in V1, which I can start using right now?

**Reply by adriangb** [reply]
Yes it’s brand new, @dmontagu just added it a couple of days ago
